TCC.Lib/Command/ProcessStartHelper.cs

```diff
@@ -0,0 +1,179 @@
+using System.Diagnostics;
+using System.Management;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace TCC.Lib.Command
+{
+    public static class ProcessStartHelper
+    {
+        public static Task<CommandResult> Run(this string command, string workingDirectory,
+            CancellationToken cancellationToken, int timeoutMinutes = 480)
+        {
+            return command.Run(workingDirectory, cancellationToken, TccProcessOutputClassifier.Instance, timeoutMinutes);
+        }
+
+        public static Task<CommandResult> Run(this string command, string workingDirectory,
+            CancellationToken cancellationToken, IProcessOutputClassifier outputClassifier, int timeoutMinutes = 480)
+        {
+            Process process = CreateProcess(command, workingDirectory);
+
+            var standardOutput = new List<string>();
+            var standardError = new List<string>();
+            var standardInfos = new List<string>();
+            var standardOutputResults = new TaskCompletionSource<List<string>>();
+            var standardErrorResults = new TaskCompletionSource<List<string>>();
+
+            process.OutputDataReceived += (sender, e) => OnOutput(e, standardOutput, standardOutputResults);
+            process.ErrorDataReceived += (sender, e) => OnError(outputClassifier, e, standardError, standardInfos, standardErrorResults);
+
+            var tcs = new TaskCompletionSource<CommandResult>();
+
+            async void OnProcessExit(object sender, EventArgs args)
+            {
+                var stdout = await standardOutputResults.Task;
+                var stderr = await standardErrorResults.Task;
+
+                var result = new CommandResult
+                {
+                    Command = command,
+                    ExitCode = process.ExitCode,
+                    IsSuccess = standardError.Count == 0,
+                    Output = string.Join(Environment.NewLine, stdout),
+                    Errors = string.Join(Environment.NewLine, stderr),
+                    Infos = standardInfos,
+                };
+
+                tcs.TrySetResult(result);
+                process.Exited -= OnProcessExit;
+            }
+
+            process.Exited += OnProcessExit;
+
+            using (cancellationToken.Register(() =>
+            {
+                tcs.TrySetCanceled();
+                process.Exited -= OnProcessExit;
+                KillProcessAndChilds(process);
+            }))
+            {
+                try
+                {
+                    if (process.Start() == false)
+                    {
+                        tcs.TrySetException(new InvalidOperationException("Failed to start process"));
+                        process.Exited -= OnProcessExit;
+                    }
+                    else
+                    {
+                        if (process.WaitForExit((int)Math.Min(TimeSpan.FromMinutes(timeoutMinutes).TotalMilliseconds, int.MaxValue)) == false)
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Remove the unnecessary Boolean literal(s). [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=csharpsquid%3AS1125)